File: CS/E2876/CustomServiceClientFactory.cs
using System.ServiceModel;
using DevExpress.Xpf.Printing.Service;
using DevExpress.XtraReports.Service;

namespace E2876
{
    public class CustomServiceClientFactory : ReportServiceClientFactory
    {
        public string EndpointConfigurationName { get; set; }

        public override IReportServiceClient Create(EndpointAddress serviceAddress)
        {
            var channelFactory = new ChannelFactory<IAsyncReportService>(EndpointConfigurationName);
            var channel = channelFactory.CreateChannel();
            return CreateClient(channel);
        }
    }
}

File: VB/E2876/CustomServiceClientFactory.vb
Imports Microsoft.VisualBasic
Imports System.ServiceModel
Imports DevExpress.Xpf.Printing.ServiceModel
' ...

Namespace E2876
	Public Class CustomServiceClientFactory
		Inherits ReportServiceClientFactory
		Private privateEndpointConfigurationName As String
		Public Property EndpointConfigurationName() As String
			Get
				Return privateEndpointConfigurationName
			End Get
			Set(ByVal value As String)
				privateEndpointConfigurationName = value
			End Set
		End Property

		Public Overrides Function Create(ByVal serviceAddress As EndpointAddress) As IReportServiceClient
			Dim channelFactory = New ChannelFactory(Of IAsyncReportService)(EndpointConfigurationName)
			Dim channel = channelFactory.CreateChannel()
			Return CreateClient(channel)
		End Function
	End Class
End Namespace
